src/hal/src/pso/descriptor.rs

```diff
@@ -67,6 +67,91 @@ pub enum DescriptorType {
     InputAttachment = 10,
 }
 
+impl std::convert::From<DescriptorType> for RawDescriptorType {
+    fn from(ty: DescriptorType) -> Self {
+        match ty {
+            DescriptorType::Sampler => RawDescriptorType::Sampler,
+            DescriptorType::CombinedImageSampler => RawDescriptorType::CombinedImageSampler,
+            DescriptorType::Image { ty } => match ty {
+                ImageDescriptorType::Storage => RawDescriptorType::StorageImage,
+                ImageDescriptorType::Sampled => RawDescriptorType::SampledImage,
+            }
+            DescriptorType::Buffer { access, format } => match access {
+                BufferDescriptorAccess::Storage => match format {
+                    BufferDescriptorFormat::Dynamic => RawDescriptorType::StorageBufferDynamic,
+                    BufferDescriptorFormat::Structured => RawDescriptorType::StorageBuffer,
+                    BufferDescriptorFormat::Texel => RawDescriptorType::StorageTexelBuffer,
+                },
+                BufferDescriptorAccess::Uniform => match format {
+                    BufferDescriptorFormat::Dynamic => RawDescriptorType::UniformBufferDynamic,
+                    BufferDescriptorFormat::Structured => RawDescriptorType::UniformBuffer,
+                    BufferDescriptorFormat::Texel => RawDescriptorType::UniformTexelBuffer,
+                },
+            }
+            DescriptorType::InputAttachment => RawDescriptorType::InputAttachment,
+        }
+    }
+}
+
+/// Access type of a buffer.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum BufferDescriptorAccess {
```

Comment:
let's rename it to `BufferDescriptorType` for the following reasons:
  - uniform/storage are not exactly different as read/write, since one can have a read-only storage. Uniform buffers are different types of bindings
  - consistency with `ImageDescriptorType`
  - avoid confusion with `buffer::Access` type